sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSqlBeamTranslationUtils.java

```diff
@@ -184,6 +188,19 @@ private static Value beamLogicalObjectToZetaSqlValue(Object object, String ident
       } else { // input type
         return Value.createTimeValue(CivilTimeEncoder.encodePacked64TimeNanos((LocalTime) object));
       }
+    } else if (SqlTypes.DATETIME.getIdentifier().equals(identifier)) {
+      // DateTime value
+      LocalDateTime datetime;
+      if (object instanceof Row) { // base type
+        datetime =
+            LocalDateTime.of(
+                LocalDate.ofEpochDay(((Row) object).getValue("Date")),
+                LocalTime.ofNanoOfDay(((Row) object).getValue("Time")));
+      } else { // input type
+        datetime = (LocalDateTime) object;
+      }
+      return Value.createDatetimeValue(
```

Comment:
Yes. It would be far more convenient to have a method creating a ZetaSQL `DateTime` value directly based on Java `LocalDateTime` object. Currently we have to extract `bitFieldDatetimeSeconds` and `nanos` from `LocalDateTime` object first.

Reply:
I created a JIRA for this: https://issues.apache.org/jira/browse/BEAM-10611

Please add a TODO here such that we can make improvement on this in the future.

Reply:
Ack.

Reply:
Done.